sat_vapor_pres/sat_vapor_pres_k.F90

```diff
@@ -1148,580 +2053,1478 @@ end subroutine compute_mrs_k_0d
+
+   select type (temp)
+   type is (real(kind=r4_kind))
+     select type (esat)
+     type is (real(kind=r4_kind))
+       select type (desat)
+       type is (real(kind=r4_kind))
+         valid_types = .true.
+       end select
+     end select
+   type is (real(kind=r8_kind))
+     select type (esat)
+     type is (real(kind=r8_kind))
+       select type (desat)
+       type is (real(kind=r8_kind))
+         valid_types = .true.
+       end select
+     end select
+   end select
+   if ( .not. valid_types ) then
+     call error_mesg ('sat_vapor_pres_k_mod::lookup_es_des_k_1d', 'unsupported kind', FATAL)
+   end if
 
    nbad = 0
-   do i = 1, size(temp,1)
-     tmp = temp(i)-tminl
-     ind = int(dtinvl*(tmp+tepsl))
-     if (ind < 0 .or. ind >= table_siz)  then
-       nbad = nbad+1
-     else
-       del = tmp-dtres*real(ind)
-       esat(i) = TABLE(ind+1) + &
-                   del*(DTABLE(ind+1) + del*D2TABLE(ind+1))
-       desat(i) = DTABLE(ind+1) + 2.*del*D2TABLE(ind+1)
-     endif
-   enddo
+
+   select type (temp)
+   type is (real(kind=r4_kind))
+     select type (esat)
+     type is (real(kind=r4_kind))
+       select type (desat)
+       type is (real(kind=r4_kind))
+         do i = 1, size(temp,1)
+           tmp = temp(i)-tminl
+           ind = int(dtinvl*(tmp+tepsl))
+           if (ind < 0 .or. ind >= table_siz)  then
+             nbad = nbad+1
+           else
+             del = tmp-dtres*real(ind)
+             esat(i) = TABLE(ind+1) + del*(DTABLE(ind+1)+del*D2TABLE(ind+1))
+             desat(i) = DTABLE(ind+1) + 2.*del*D2TABLE(ind+1)
+           endif
+         enddo
+       end select
+     end select
+   type is (real(kind=r8_kind))
+     select type (esat)
+     type is (real(kind=r8_kind))
+       select type (desat)
+       type is (real(kind=r8_kind))
+         do i = 1, size(temp,1)
+           tmp = temp(i)-tminl
+           ind = int(dtinvl*(tmp+tepsl))
+           if (ind < 0 .or. ind >= table_siz)  then
+             nbad = nbad+1
+           else
+             del = tmp-dtres*real(ind)
+             esat(i) = TABLE(ind+1) + del*(DTABLE(ind+1)+del*D2TABLE(ind+1))
+             desat(i) = DTABLE(ind+1) + 2.*del*D2TABLE(ind+1)
+           endif
+         enddo
+       end select
+     end select
+   end select
 
  end subroutine lookup_es_des_k_1d
 
 !#######################################################################
 
  subroutine lookup_es_des_k_0d (temp, esat, desat, nbad)
- real, intent(in)     :: temp
- real, intent(out)    :: esat, desat
+ class(*), intent(in)     :: temp
+ class(*), intent(out)    :: esat, desat
  integer, intent(out) :: nbad
 
  real    :: tmp, del
  integer :: ind
 
    nbad = 0
-   tmp = temp-tminl
+
+   select type (temp)
+   type is (real(kind=r4_kind))
+     tmp = temp-tminl
+   type is (real(kind=r8_kind))
+     tmp = temp-tminl
+   class default
+     call error_mesg ('sat_vapor_pres_k_mod::lookup_es_des_k_0d', 'unsupported kind', FATAL)
```

Comment:
This error message should be more descriptive.  *"temp is not one of the supported types of real(kind=4) or real(kind=8)."*